chromadb/db/impl/grpc/server.py

```diff
@@ -317,6 +320,19 @@ def CreateCollection(
             version=0,
         )
         collections[request.id] = new_collection
+        
+        # Create segments for the collection
+        for segment_proto in request.segments:
+            segment = from_proto_segment(segment_proto)
+            response = self.CreateSegmentHelper(segment)
+            if response.status.code != 200:
+                return CreateCollectionResponse(
```

Comment:
Note: I am deprecating the status field in the proto response here https://github.com/chroma-core/chroma/pull/2896